crates/biome_js_analyze/src/lint/nursery/use_spread.rs

```diff
@@ -0,0 +1,149 @@
+use crate::JsRuleAction;
+use biome_analyze::{
+    Ast, FixKind, Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsCallArgument, AnyJsMemberExpression, JsCallExpression, JsSyntaxKind, T,
+};
+use biome_rowan::{AstNode, AstSeparatedList, BatchMutationExt};
+use biome_rule_options::use_spread::UseSpreadOptions;
+
+declare_lint_rule! {
+    /// Enforce the use of the spread operator over `.apply()`.
+    ///
+    /// The `apply()` method is used to call a function with a given `this` value and arguments provided as an array.
+    /// The spread operator `...` can be used to achieve the same result, which is more concise and easier to read.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// foo.apply(null, args);
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// foo.apply(undefined, args);
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// obj.foo.apply(obj, args);
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// foo(...args);
+    ///
+    /// obj.foo(...args);
+    ///
+    /// foo.apply(obj, [1, 2, 3]);
+    ///
+    /// foo.apply(null, [1, 2, 3]);
+    /// ```
+    ///
+    pub UseSpread {
+        version: "next",
+        name: "useSpread",
+        language: "js",
+        sources: &[RuleSource::Eslint("prefer-spread").same()],
+        recommended: true,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+impl Rule for UseSpread {
+    type Query = Ast<JsCallExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = UseSpreadOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let callee = node.callee().ok()?;
+
+        let member_expr = AnyJsMemberExpression::cast(callee.into_syntax())?;
+        if member_expr.member_name()?.text() != "apply" {
+            return None;
+        }
+
+        let arguments = node.arguments().ok()?.args();
+        if arguments.len() != 2 {
+            return None;
+        }
+
+        let first_arg = arguments.first()?.ok()?;
+        let this_arg = first_arg.as_any_js_expression()?;
+
+        let second_arg = arguments.last()?.ok()?;
+        let spread_candidate = second_arg.as_any_js_expression()?;
+
+        // The rule should not flag `.apply()` calls where the second argument is an array literal.
+        if spread_candidate.syntax().kind() == JsSyntaxKind::JS_ARRAY_EXPRESSION {
+            return None;
+        }
+
+        let applied_object = member_expr.object().ok()?;
+
+        let is_this_correct = if let Some(object_member) =
+            AnyJsMemberExpression::cast(applied_object.clone().into_syntax())
+        {
+            object_member.object().ok()?.syntax().text_trimmed() == this_arg.syntax().text_trimmed()
```

Comment:
What is this comparison doing? At first glance, I would say it's incorrect because we're comparing the text of two nodes, which is something we never do because it causes many errors. Calling `text_trimmed` on a node removes the trivia that belong to the node, _but it doesn't remove the trivia that belong to all the descendants._

We usually query **the tokens we want to compare**, so they don't contain trivia